registry/npm.js

```diff
@@ -1,9 +1,31 @@
-const gop = require('get-owned-packages')
+const fetch = require('npm-registry-fetch')
 
-function NpmRegistry () {}
+class NpmRegistry {
+  constructor () {
+    this.constants = {
+      registry: 'https://registry.npmjs.org/',
+      readWrite: 'read-write',
+      write: 'write'
+    }
+  }
 
-NpmRegistry.prototype.getOwnedPackages = async function getOwnedPackages (token) {
-  return gop.getOwnedPackages(token)
+  async getUsername ({ readOnlyToken }) {
+    const opts = { registry: this.constants.registry, token: readOnlyToken }
+
+    // will throw in the case of an invalid token
+    const userData = await fetch.json('/-/npm/v1/user', opts)
```

Comment:
is there a situation in which this returns data but not name?

Reply:
afaik: nope; if token is valid it's gotta be tied to some user and all NPM users have names. invalid token throws